File: kgparserSrv/src/main/java/ru/klavogonki/kgparser/DictionaryMode.java
package ru.klavogonki.kgparser;

import ru.klavogonki.common.StandardDictionary;

public enum DictionaryMode {

    normal,

    noerror,

    sprint,

    marathon,

    ;

    public static DictionaryMode getDictionaryMode(String dictionaryCode) {
        if (!Dictionary.isStandard(dictionaryCode)) {
            return normal;
        }

        StandardDictionary standardDictionary = StandardDictionary.getByKlavogonkiName(dictionaryCode);

        switch (standardDictionary) {
            case normal:
            case chars:
            case abra:
            case referats:
            case digits:
                return normal;

            case sprint:
                return sprint;

            case noerror:
                return noerror;

            case marathon:
                return marathon;

            default:
                throw new IllegalArgumentException(String.format("Unknown standard dictionary code: %s", dictionaryCode));
        }
    }
}

File: kgCommon/src/main/kotlin/ru/klavogonki/common/DictionaryMode.kt
package ru.klavogonki.common

import ru.klavogonki.common.DictionaryUtils.isStandard

enum class DictionaryMode(

    /**
     * Имя в API клавогонок. Все буквы в нижнем регистре.
     */
    @JvmField val klavogonkiName: String
) {
    normal("normal"),

    noerror("noerror"),

    sprint("sprint"),

    marathon("marathon"),
    ;

    companion object {

        @JvmStatic
        fun getDictionaryMode(dictionaryCode: String): DictionaryMode {
            if (!isStandard(dictionaryCode)) {
                return normal
            }

            val standardDictionary = StandardDictionary.getByKlavogonkiName(dictionaryCode)
            return standardDictionary.dictionaryMode
        }
    }
}
